### ISSUE TITLE
Connection gets aborted after upgrading from postgresql 9.3 to 9.4

### ISSUE THREAD

**Issue by mitalpritmani** [EnterpriseDB/mongo_fdw#15]
I had PostgreSQL 9.3 but as I wanted to use jsonb, I upgraded to 9.4 and reinstalled mongo_fdw extension but now it has stopped working.

I can create extension and also user mapping but when after creating foreign table, I fire select query, it aborts the connection to my pg database and exits even from psql connection.

Can you please help me what could have gone wrong?

**Comment by ahsanhadi** [comment]
On Wed, Jan 7, 2015 at 2:52 PM, mitalpritmani <notifications@github.com>
wrote:

> I had PostgreSQL 9.3 but as I wanted to use jsonb, I upgraded to 9.4 and
> reinstalled mongo_fdw extension but now it has stopped working.
>
> I can create extension and also user mapping but when after creating
> foreign table, I fire select query, it aborts the connection to my pg
> database and aborts it.
>
> Can you please help me what could have gone wrong?
>

Sure. I will try to reproduce the issue and revert back to you.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EnterpriseDB/mongo_fdw/issues/15>.
>



-- 
Ahsan Hadi
Snr Director Product Development
EnterpriseDB Corporation
The Enterprise Postgres Company

Phone: +92-51-8358874
Mobile: +92-333-5162114

Website: www.enterprisedb.com
EnterpriseDB Blog: http://blogs.enterprisedb.com/
Follow us on Twitter: http://www.twitter.com/enterprisedb

This e-mail message (and any attachment) is intended for the use of the
individual or entity to whom it is addressed. This message contains
information from EnterpriseDB Corporation that may be privileged,
confidential, or exempt from disclosure under applicable law. If you are
not the intended recipient or authorized to receive this for the intended
recipient, any use, dissemination, distribution, retention, archiving, or
copying of this communication is strictly prohibited. If you have received
this e-mail in error, please notify the sender immediately by reply e-mail
and delete this message.

**Comment by mitalpritmani** [comment]
Thanks ahsanhadi. But issue got resolved.

I'm not sure what exactly was the problem. But I removed all the extensions from my postgresql 9.4 and followed all the steps again. The reason why I did was, earlier I had got the error while creating extension, "the version of mongo_fdw library mismatches the version of postgres". So following the steps of installing mongo_fdw extension created that extension in my extension folder of postgres. Then restarting the postgres resolved the issue.

The only problem is, I have lost all the extensions from my postgres which were created when I had followed steps previously. But I'll see how to get them back. For now, I can get data from mongo db.
